### PR TITLE
[INTEL MKL] Integrated MKL to TF conversion op with MKL-DNN v1.x

### PR DESCRIPTION
**Pull request by bhavani-subramanian**

### CODE REVIEW

File: tensorflow/core/kernels/mkl_tfconv_op.h [hunk 1]
```diff
@@ -101,13 +110,34 @@ class MklToTfOp : public OpKernel {
...
-        CHECK_EQ(input.CheckReorderToOpMem(output_tf_pd, output_tensor), true);
+      // Do we need to reorder MKL layout into TensorFlow layout?
+      if (input.IsReorderNeeded(OUTPUT_TF_MD)) {
+#ifdef ENABLE_MKLDNN_V1
+        std::vector<primitive> net;
+        std::vector<MemoryArgsMap> net_args;
+        // Insert reorder between MKL layout and TensorFlow layout.
+        // TODO(bhavanis): Refactor CheckReorderToOpMem() to directly insert
+        // the reorder primitive
+        bool status = input.CheckReorderToOpMem(output_tf_md, output_tensor,
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_tfconv_op.h]: The statement can be inside `OP_REQUIRES`.  Same for the one below.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/core/kernels/mkl_tfconv_op.h (hunk 1 above)
Comment (bhavani-subramanian) [reply tensorflow/core/kernels/mkl_tfconv_op.h]: Done

**Comment by bhavani-subramanian** [comment]
@penpornk I have addressed your review comments. Please take a look. Thank you!

File: tensorflow/core/kernels/mkl_tfconv_op.h [hunk 2]
```diff
@@ -37,13 +37,17 @@ limitations under the License.
 using mkldnn::stream;
 
 namespace tensorflow {
+
 #ifdef ENABLE_MKLDNN_V1
 #define ENGINE_CPU engine::kind::cpu
+#define NET_ARGS_PTR &net_args
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_tfconv_op.h]: `NET_ARGS_PTR` is not used in this file.

File: tensorflow/core/kernels/mkl_tfconv_op.h [hunk 3]
```diff
@@ -110,32 +114,16 @@ class MklToTfOp : public OpKernel {
...
-        // Insert reorder between MKL layout and TensorFlow layout.
-        // TODO(bhavanis): Refactor CheckReorderToOpMem() to directly insert
-        // the reorder primitive
-        bool status = input.CheckReorderToOpMem(output_tf_md, output_tensor,
-                                                net, net_args, cpu_engine);
-        OP_REQUIRES(
-            context, status,
-            errors::Internal("MklToTfOp: Failed to create input reorder"));
-
-        ExecutePrimitive(net, &net_args, cpu_engine);
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_tfconv_op.h]: `ExecutePrimitive` is not called anymore. Is this intentional?

**Review by penpornk — COMMENTED** [review]
Thank you for the edits! I have a few questions please.

File: tensorflow/core/kernels/mkl_tfconv_op.h (hunk 2 above)
Comment (bhavani-subramanian) [reply tensorflow/core/kernels/mkl_tfconv_op.h]: Yes, you're right. I removed this macro.

File: tensorflow/core/kernels/mkl_tfconv_op.h (hunk 3 above)
Comment (bhavani-subramanian) [reply tensorflow/core/kernels/mkl_tfconv_op.h]: Actually, this version of `CheckReorderToOpMem()` both creates and executes the reorder primitive: i.e. https://github.com/tensorflow/tensorflow/blob/063a8303f69645e1da610ef8a5bbfcfc782e42bb/tensorflow/core/util/mkl_util.h#L1680 calls https://github.com/tensorflow/tensorflow/blob/063a8303f69645e1da610ef8a5bbfcfc782e42bb/tensorflow/core/util/mkl_util.h#L1613 which in turn executes reorder.

Comment (penpornk) [reply tensorflow/core/kernels/mkl_tfconv_op.h]: Got it. Thank you for the explanation! :)

**Review by penpornk — APPROVED** [review]

**Pull request merged**